### DISCUSSION TITLE
Documentation for imap_filter

### DISCUSSION THREAD

**Discussion by allgoewer** [emersion/maddy#368] [Q&A] [answered]
Hi all,

I'm having a hard time figuring out where to put the imap_filter directive in the default maddy.conf .

The documentation isn't really clear on that (at least to me) (see [Doc](https://maddy.email/man/_generated_maddy-imap.5/)) .

Can anyone help me out?

**Comment by foxcpp** [comment, marked as answer]
Good catch. Documentation for IMAP metadata filtering is a bit lacking.

[maddy-storage](https://maddy.email/man/_generated_maddy-storage.5/) specifies imap_filters directive that should be set in storage.imapsql block.

E.g. like this:
```
storage.imapsql local_mailboxes {
    driver sqlite3
    dsn imapsql.db
    
    imap_filter {
      command /var/lib/maddy/sieve.sh {account_name}
    }
}
```

It might be counter-intuitive (why not `imap` endpoint block?) but makes more sense if you consider that IMAP endpoint has nothing to do with message delivery.

**Comment by allgoewer** [comment]
I already tried that

```
storage.imapsql local_mailboxes {
    driver sqlite3
    dsn imapsql.db

    imap_filters {
        command /etc/maddy/sieve-{account_name}.sh
    }
}
```

leads to

```
Jul 08 17:24:40 ganymed maddy[9903]: /etc/maddy/maddy.conf:52: unexpected dir
ective: imap_filters
```

**Reply by foxcpp** [reply]
Oh, sorry, apparently there is a typo in the documentation and the actual directive name is `imap_filter`.
https://github.com/foxcpp/maddy/blob/27f468dd2e6fab002e5f00e135129eceed28a164/internal/storage/imapsql/imapsql.go#L269

**Reply by allgoewer** [reply]
That works, thanks!

I think what confused me was the documentation referring to the directive as `imap.filter.command`.

I originally thought it should be going into the `imap` directive.

**Reply by foxcpp** [reply]
I improved maddy-imap text: https://maddy.email/man/_generated_maddy-imap.5/
I hope it makes it more clear.

Also note that placeholders are not expanded in command name to prevent command injection attacks (not relevant if you use just {account_name} but other placeholders may be more dangerous).
